File: tests/integration/fixtures/typescript-project/src/parser.ts
/**
 * Parser module — has unsorted imports for normalize-imports tests.
 */
import { formatNumber } from './utils';
import { flatten } from './utils';
import { formatSum } from './formatter';
import { add } from './calculator';

export function parseInt(value: unknown): number | null {
  const n = Number(value);
  if (!Number.isInteger(n)) return null;
  return n;
}

export function parseFloat(value: unknown): number | null {
  const n = Number(value);
  if (isNaN(n)) return null;
  return n;
}

export function parseList(value: string, separator: string = ','): string[] {
  if (!value) return [];
  return value.split(separator).map((v) => v.trim()).filter((v) => v.length > 0);
}

export function parseNumberList(value: string): number[] {
  const tokens = parseList(value);
  return tokens.map((t) => Number(t)).filter((n) => !isNaN(n));
}

File: tests/integration/fixtures/python-project/src/parser.py
"""Parser module — has intentionally unsorted and duplicate imports for normalize-imports tests."""
import sys
import os
import json
import math
import os  # duplicate import for normalize-imports testing
import sys  # duplicate import for normalize-imports testing

from src.utils import flatten, format_number
from src.calculator import add


def parse_int(value):
    """Parse value as integer, returning None on failure."""
    try:
        return int(value)
    except (TypeError, ValueError):
        return None


def parse_float(value):
    """Parse value as float, returning None on failure."""
    try:
        return float(value)
    except (TypeError, ValueError):
        return None


def parse_list(value, separator=","):
    """Split value by separator and return list of stripped tokens."""
    if not value:
        return []
    parts = [v.strip() for v in value.split(separator)]
    return flatten([[p] for p in parts if p])


def parse_json_safe(text):
    """Parse JSON text; return None on error."""
    try:
        return json.loads(text)
    except (json.JSONDecodeError, TypeError):
        return None
